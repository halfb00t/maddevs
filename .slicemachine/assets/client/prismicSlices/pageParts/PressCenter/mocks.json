[
  {
    "variation": "default-slice",
    "name": "Default slice",
    "slice_type": "press_center",
    "items": [
      {
        "image": {
          "dimensions": {
            "width": 900,
            "height": 500
          },
          "alt": "Placeholder image",
          "copyright": null,
          "url": "https://images.unsplash.com/photo-1579931794097-0ad001e51edb?w=900&h=500&fit=crop"
        },
        "title": "maximize out-of-the-box portals",
        "description": "monetize virtual convergence",
        "link": {
          "link_type": "Web",
          "url": "http://google.com"
        },
        "date": "2013-03-17",
        "linkLabel": "benchmark frictionless models",
        "linkedCompanyLogo": {
          "dimensions": {
            "width": 900,
            "height": 500
          },
          "alt": "Placeholder image",
          "copyright": null,
          "url": "https://images.unsplash.com/photo-1494173853739-c21f58b16055?w=900&h=500&fit=crop"
        }
      },
      {
        "image": {
          "dimensions": {
            "width": 900,
            "height": 500
          },
          "alt": "Placeholder image",
          "copyright": null,
          "url": "https://images.unsplash.com/photo-1600861194802-a2b11076bc51?w=900&h=500&fit=crop"
        },
        "title": "enhance strategic synergies",
        "description": "morph synergistic functionalities",
        "link": {
          "link_type": "Web",
          "url": "http://twitter.com"
        },
        "date": "2018-09-23",
        "linkLabel": "deploy world-class e-services",
        "linkedCompanyLogo": {
          "dimensions": {
            "width": 900,
            "height": 500
          },
          "alt": "Placeholder image",
          "copyright": null,
          "url": "https://images.unsplash.com/photo-1606248897732-2c5ffe759c04?w=900&h=500&fit=crop"
        }
      }
    ],
    "primary": {
      "title": "deploy dot-com architectures",
      "description": "revolutionize seamless initiatives",
      "date": "2014-12-22",
      "link": {
        "link_type": "Web",
        "url": "https://prismic.io"
      },
      "linkLabel": "deliver mission-critical users",
      "image": {
        "dimensions": {
          "width": 900,
          "height": 500
        },
        "alt": "Placeholder image",
        "copyright": null,
        "url": "https://images.unsplash.com/photo-1515378960530-7c0da6231fb1?w=900&h=500&fit=crop"
      },
      "linkedCompanyLogo": {
        "dimensions": {
          "width": 900,
          "height": 500
        },
        "alt": "Placeholder image",
        "copyright": null,
        "url": "https://images.unsplash.com/photo-1493119508027-2b584f234d6c?w=900&h=500&fit=crop"
      }
    }
  }
]