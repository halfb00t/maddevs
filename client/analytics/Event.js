/* eslint-disable no-console, class-methods-use-this, no-underscore-dangle */
import uniqid from '~/helpers/generatorUid'

const LOCAL_STORAGE_KEYS = {
  ID: 'GA_USER_ID',
  TYPE: 'GA_USER_TYPE',
}

export const addUserType = type => {
  // TYPE: 'hr_candidate' | 'lead' | 'download_ebook' | undefined
  localStorage.setItem(LOCAL_STORAGE_KEYS.TYPE, type)
}

export class AnalyticsEvent {
  constructor(action, strict = true) {
    if (!action) {
      throw new Error(`Event action is missing for ${this.action}, please add an action for object`)
    }
    this.strict = strict
    this.action = action
    this.properties = {}
    this._applyUser()
  }

  _applyUser() {
    if ('window' in global) {
      const existId = localStorage.getItem(LOCAL_STORAGE_KEYS.ID)
      const type = localStorage.getItem(LOCAL_STORAGE_KEYS.TYPE)

      const userId = existId || uniqid()

      localStorage.setItem(LOCAL_STORAGE_KEYS.ID, userId)

      this.properties.user_id = userId
      this.properties.user_type = type
    }
  }

  _handleError(message) {
    if (this.strict) throw new Error(message)
    console.error(message)
  }

  setCategory(category) {
    if (!category) {
      this._handleError('Event category is required')
    }
    this.properties.event_category = category
    return this
  }

  setField(name, value) {
    if (!name || !(value !== undefined && value !== null)) {
      this._handleError('AnalyticsEvent.setField() method is failed. Please add name or values params')
    }
    this.properties[name] = value
    return this
  }

  _setPath() {
    this.properties.path = window.location.pathname
  }

  _collectGoogleAnalyticsKeys() {
    const keys = window.dataLayer
      ?.filter(dataLayerValue => dataLayerValue[0] === 'config' && dataLayerValue.length >= 3)
      .map(dataLayerValue => dataLayerValue[1]) || []

    return Array.from(new Set(keys))
  }

  _log(analyticsKeys) {
    const msg = [
      '<------- ANALYTICS EVENT SENDING -------->',
      `EVENT: ${this.action}`,
      `PROPS: ${JSON.stringify(this.properties)}`,
      `ANALYTICS: ${JSON.stringify(analyticsKeys)}`,
    ]

    console.log(msg.join('\n'))
  }

  send() {
    if (process.env.domain === 'https://maddevs.io' || process.env.domain === 'https://maddevs.co') {
      this._setPath()
      this._applyUser()

      const analyticsKeys = this._collectGoogleAnalyticsKeys()

      if (process.env.NODE_ENV === 'development') {
        this._log(analyticsKeys)
      }

      analyticsKeys.forEach(analyticsId => {
        const properties = { ...this.properties, send_to: analyticsId }
        try {
          window.gtag('event', this.action, properties)
        } catch (error) {
          this._handleError(error)
        }
      })
    }
  }
}
