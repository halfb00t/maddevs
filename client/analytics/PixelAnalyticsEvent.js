/* eslint-disable no-console, class-methods-use-this, no-underscore-dangle */
import uniqid from '~/helpers/generatorUid'

const LOCAL_STORAGE_KEYS = {
  ID: 'GA_USER_ID',
  TYPE: 'GA_USER_TYPE',
}

export const TRACK_TYPES = {
  TRACK: 'track',
  CUSTOM_TRACK: 'trackCustom',
}

export const addUserType = type => {
  // TYPE: 'hr_candidate' | 'lead' | 'download_ebook' | undefined
  localStorage.setItem(LOCAL_STORAGE_KEYS.TYPE, type)
}

export class PixelAnalyticsEvent {
  constructor(action, trackType = TRACK_TYPES.TRACK, strict = true) {
    if (!action) {
      throw new Error(`Event action is missing for ${this.action}, please add an action for object`)
    }
    this.strict = strict
    this.action = action
    this.properties = {}
    this.trackType = trackType
    this._applyUser()
  }

  _applyUser() {
    if ('window' in global) {
      const existId = localStorage.getItem(LOCAL_STORAGE_KEYS.ID)
      const type = localStorage.getItem(LOCAL_STORAGE_KEYS.TYPE)

      const userId = existId || uniqid()

      localStorage.setItem(LOCAL_STORAGE_KEYS.ID, userId)

      this.properties.user_id = userId
      this.properties.user_type = type
    }
  }

  _handleError(message) {
    if (this.strict) throw new Error(message)
    console.error(message)
  }

  setCategory(category) {
    if (!category) {
      this._handleError('Event category is required')
    }
    this.properties.event_category = category
    return this
  }

  setField(name, value) {
    if (!name || !(value !== undefined && value !== null)) {
      this._handleError('AnalyticsEvent.setField() method is failed. Please add name or values params')
    }
    this.properties[name] = value
    return this
  }

  _setPath() {
    this.properties.path = window.location.pathname
  }

  _collectGoogleAnalyticsKeys() {
    const keys = window.dataLayer
      ?.filter(dataLayerValue => dataLayerValue[0] === 'config' && dataLayerValue.length >= 3)
      .map(dataLayerValue => dataLayerValue[1]) || []

    return Array.from(new Set(keys))
  }

  _log(analyticsKeys) {
    const msg = [
      '<------- ANALYTICS EVENT SENDING -------->',
      `EVENT: ${this.action}`,
      `PROPS: ${JSON.stringify(this.properties)}`,
      `ANALYTICS: ${JSON.stringify(analyticsKeys)}`,
    ]

    console.log(msg.join('\n'))
  }

  send() {
    if (window.location.hostname === 'maddevs.io'
      || window.location.hostname === 'maddevs.co'
      || window.location.origin === 'https://maddevs.co'
      || window.location.origin === 'https://maddevs.io'
    ) {
      this._setPath()
      this._applyUser()

      const analyticsKeys = this._collectGoogleAnalyticsKeys()

      if (process.env.NODE_ENV === 'development') {
        this._log(analyticsKeys)
      }

      analyticsKeys.forEach(analyticsId => {
        const properties = { ...this.properties, send_to: analyticsId }
        try {
          window.fbq(this.trackType, this.action, properties)
        } catch (error) {
          this._handleError(error)
        }
      })
    }
  }
}
