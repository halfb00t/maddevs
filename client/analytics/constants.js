export const CONTENT_GROUPS = [
  { url: ['http://maddevs.io/'], name: 'home_page' },
  { url: ['https://maddevs.io/case-studies/'], name: 'case_studies' },
  { url: ['https://maddevs.io/case-studies/clutch/'], name: 'case' },
  { url: ['https://maddevs.io/blog/'], name: 'blog' },
  { url: ['https://maddevs.io/ebooks/pricing-strategies/'], name: 'pricing-strategies' },
  { url: ['https://maddevs.io/customer-university/custom-software-development-pricing-strategies/'], name: 'cu_post' },
  { url: ['https://maddevs.io/blog/automation-of-customer-service/'], name: 'blog_post' },
  { url: ['https://maddevs.io/open-source/'], name: 'open_source' },
  {
    url: [
      'https://maddevs.io/services/',
      'https://maddevs.io/software-testing/',
    ],
    name: 'services',
  },
  {
    url: [
      'https://maddevs.io/development-process/',
      'https://maddevs.io/delivery-models/',
    ],
    name: 'approach',
  },
  { url: ['https://maddevs.io/team/'], name: 'team' },
  { url: ['https://maddevs.io/careers/'], name: 'careers' },
  { url: ['https://maddevs.io/ru/careers/senior-golang/'], name: 'job_post' },
  { url: ['https://maddevs.io/contact-us/'], name: 'contact_us' },
  { url: ['https://maddevs.io/tag/frontend-development/'], name: 'tag' },
  { url: ['https://maddevs.io/author/denis-grushkin/'], name: 'author' },
  {
    url: [
      'https://maddevs.io/privacy/',
      'https://maddevs.io/sustainability-policy/',
      'https://maddevs.io/gdpr/',
      'https://maddevs.io/nda/',
      'https://maddevs.io/faq/',
    ],
    name: 'policy',
  },
  {
    url: [
      'https://maddevs.io/transportation/',
      'https://maddevs.io/blockchain/',
    ],
    name: 'industries',
  },
]

export const EVENT_CATEGORIES = {
  CLICK: 'click',
  SCROLL: 'scroll',
  SUBMIT: 'submit',
}

export const ANALYTICS_IDS = {
  GA4: process.env.analytics4Key,
  UNIVERSAL: process.env.analyticsUniversalKey,
}
