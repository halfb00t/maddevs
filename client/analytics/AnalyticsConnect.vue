<template>
  <div />
</template>

<script>
import { ANALYTICS_IDS } from '@/analytics/constants'

export default {
  data() {
    return {
      ANALYTICS_IDS,
    }
  },

  mounted() {
    window.dataLayer = window.dataLayer || []

    function gtag(...args) {
      window.dataLayer.push(args)
    }

    window.gtag = gtag

    window.gtag('js', new Date())
    window.gtag('config', this.ANALYTICS_IDS.GA4, { debug_mode: true })
    window.gtag('config', this.ANALYTICS_IDS.UNIVERSAL, { debug_mode: true })
  },
}
</script>
