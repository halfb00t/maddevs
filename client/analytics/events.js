import { AnalyticsEvent } from './Event'
import { EVENT_CATEGORIES } from './constants'

export const headerClickEvent = new AnalyticsEvent('click').setCategory(EVENT_CATEGORIES.CLICK)

export const phoneClickEvent = new AnalyticsEvent('phone_click').setCategory(EVENT_CATEGORIES.CLICK)

export const emailClickEvent = new AnalyticsEvent('email_click').setCategory(EVENT_CATEGORIES.CLICK)

export const contactMeClickEvent = new AnalyticsEvent('contact_me_click').setCategory(EVENT_CATEGORIES.CLICK)

export const contactMeSubmitEvent = new AnalyticsEvent('submit_contact_me').setCategory(EVENT_CATEGORIES.SUBMIT)

export const cardClickEvent = new AnalyticsEvent('card_click').setCategory(EVENT_CATEGORIES.CLICK)

export const socialNetworkClickEvent = new AnalyticsEvent('social_network_click').setCategory(EVENT_CATEGORIES.CLICK)

export const footerFormSubmitEvent = new AnalyticsEvent('submit_footer_form').setCategory(EVENT_CATEGORIES.SUBMIT)

export const contactsClickEvent = new AnalyticsEvent('contacts_click').setCategory(EVENT_CATEGORIES.CLICK)

export const blockClickEvent = new AnalyticsEvent('blog_click').setCategory(EVENT_CATEGORIES.CLICK)

export const categoryPageClickEvent = new AnalyticsEvent('category_click').setCategory(EVENT_CATEGORIES.CLICK)

export const pressCenterClickEvent = new AnalyticsEvent('press_click').setCategory(EVENT_CATEGORIES.CLICK)

export const linkedinUserClickEvent = new AnalyticsEvent('linkedin_user_click').setCategory(EVENT_CATEGORIES.CLICK)

export const careersClickEvent = new AnalyticsEvent('careers_click').setCategory(EVENT_CATEGORIES.CLICK)

export const hrSiteClickEvent = new AnalyticsEvent('hr_site_click').setCategory(EVENT_CATEGORIES.CLICK)

export const githubClickEvent = new AnalyticsEvent('github_click').setCategory(EVENT_CATEGORIES.CLICK)

export const writeUsClickEvent = new AnalyticsEvent('write_us_click').setCategory(EVENT_CATEGORIES.CLICK)

export const submitCvFormEvent = new AnalyticsEvent('submit_cv').setCategory(EVENT_CATEGORIES.SUBMIT)

export const pricingStrategiesEbookSubmitFormEvent = new AnalyticsEvent('submit_pricing_strategies_ebook_form').setCategory(EVENT_CATEGORIES.SUBMIT)

export const approachDPEbookSubmitFormEvent = new AnalyticsEvent('submit_approach_dp_ebook_form').setCategory(EVENT_CATEGORIES.SUBMIT)
