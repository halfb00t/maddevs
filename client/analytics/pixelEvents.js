import { EVENT_CATEGORIES } from './constants'
import { PixelAnalyticsEvent, TRACK_TYPES } from './PixelAnalyticsEvent'

export const headerClickEvent = new PixelAnalyticsEvent('click').setCategory(EVENT_CATEGORIES.CLICK)

export const phoneClickPixelEvent = new PixelAnalyticsEvent('phone_click').setCategory(EVENT_CATEGORIES.CLICK)

export const contactMeClickPixelEvent = new PixelAnalyticsEvent('contact_me').setCategory(EVENT_CATEGORIES.CLICK)

export const emailClickPixelEvent = new PixelAnalyticsEvent('email_click').setCategory(EVENT_CATEGORIES.CLICK)

export const hrSiteClickPixelEvent = new PixelAnalyticsEvent('hr_site_click').setCategory(EVENT_CATEGORIES.CLICK)

export const socialNetworkClickPixelEvent = new PixelAnalyticsEvent('social_network_click').setCategory(EVENT_CATEGORIES.CLICK)

export const writeUsClickPixelEvent = new PixelAnalyticsEvent('write_us_click').setCategory(EVENT_CATEGORIES.CLICK)

export const ebookSubmitFormPixelEvent = new PixelAnalyticsEvent('submit_ebook_form').setCategory(EVENT_CATEGORIES.SUBMIT)

export const githubClickPixelEvent = new PixelAnalyticsEvent('github_click').setCategory(EVENT_CATEGORIES.CLICK)

export const pressCenterClickPixelEvent = new PixelAnalyticsEvent('press_click').setCategory(EVENT_CATEGORIES.CLICK)

export const submitCvFormPixelEvent = new PixelAnalyticsEvent('submit_cv').setCategory(EVENT_CATEGORIES.CLICK)

export const linkedinUserClickPixelEvent = new PixelAnalyticsEvent('linkedin_user_click').setCategory(EVENT_CATEGORIES.CLICK)

export const twitterNetworkPixelEvent = new PixelAnalyticsEvent('twitter', TRACK_TYPES.CUSTOM_TRACK).setCategory(EVENT_CATEGORIES.CLICK)

export const facebookNetworkPixelEvent = new PixelAnalyticsEvent('facebook', TRACK_TYPES.CUSTOM_TRACK).setCategory(EVENT_CATEGORIES.CLICK)

export const linkedinNetworkPixelEvent = new PixelAnalyticsEvent('facebook', TRACK_TYPES.CUSTOM_TRACK).setCategory(EVENT_CATEGORIES.CLICK)
