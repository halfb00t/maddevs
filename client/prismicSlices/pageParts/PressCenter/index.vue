<template>
  <section class="section press-center-section">
    <div class="container">
      <div
        :data-aos="animation"
        class="press-center"
      >
        <!--  primary zone  -->
        <!--   \/  \/  \/   -->
        <div class="press-center-main ps-main-card">
          <div class="ps-main-card-section ps-main-card-section--left">
            <h2 class="ps-main-card__title">
              {{ title }}
            </h2>
            <div
              class="ps-main-card__description"
            >
              {{ shortDescription(description) }}
            </div>
            <div class="ps-main-card__footer">
              <span class="ps-main-card__date">{{ formattedDate(date) }}</span>
              <a
                :href="link.url"
                class="ps-main-card__link"
                target="_blank"
                @click="sendPressCenterPostEvent"
              >
                <span class="ps-main-card__link-label">
                  {{ linkLabel }}
                </span>
              </a>
            </div>
          </div>
          <div class="ps-main-card-section ps-main-card-section--right">
            <img
              v-lazy-load
              :data-src="image.url"
              :alt="image.alt"
              class="ps-main-card__image"
            >
            <img
              v-lazy-load
              :data-src="linkedCompanyLogo.url"
              :alt="linkedCompanyLogo.alt"
              class="ps-main-card__linked-logo"
            >
          </div>
        </div>
        <!--   /\  /\  /\   -->
        <!--primary zone end-->
        <!-------------------->
        <!--repeatable zone -->
        <!--   \/  \/  \/   -->
        <div class="press-center-sub ps-sub-cards">
          <div
            v-for="(item, i) in subPosts"
            :key="`slice-item-${i}`"
            class="press-center-sub-wrapper ps-sub-card-wrapper"
          >
            <div

              class="press-center-sub ps-sub-card"
            >
              <div class="ps-sub-card-section ps-sub-card-section--left">
                <img
                  v-lazy-load
                  :data-src="item.image.url"
                  :alt="item.image.alt"
                  class="ps-sub-card__image"
                >
                <img
                  v-lazy-load
                  :data-src="item.linkedCompanyLogo.url"
                  :alt="item.linkedCompanyLogo.alt"
                  class="ps-sub-card__linked-logo"
                >
              </div>
              <div class="ps-sub-card-section ps-sub-card-section--right">
                <h2 class="ps-sub-card__title">
                  {{ item.title }}
                </h2>
                <div
                  class="ps-sub-card__description"
                >
                  {{ shortDescriptionForSubPosts(item.description) }}
                </div>
                <div class="ps-sub-card__footer">
                  <span class="ps-sub-card__date">{{ formattedDate(item.date) }}</span>
                  <a
                    :href="item.link.url"
                    class="ps-sub-card__link"
                    target="_blank"
                    @click="sendPressCenterPostEvent"
                  >
                    <span class="ps-sub-card__link-label">
                      {{ item.linkLabel }}
                    </span>
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <!--    /\  /\  /\     -->
        <!--repeatable zone end-->
      </div>
    </div>
  </section>
</template>

<script>
import textEllipsis from '@/helpers/textEllipsis'
import formatDate from '@/helpers/formatDate'
import { pressCenterClickEvent } from '@/analytics/events'
import { pressCenterClickPixelEvent } from '@/analytics/pixelEvents'

export default {
  name: 'PressCenter',

  props: {
    slice: {
      type: Object,

      default() {
        return {}
      },
    },
  },

  data() {
    return {
      title: this.slice?.primary?.title,
      description: this.slice?.primary?.description,
      date: this.slice?.primary?.date,
      link: this.slice?.primary?.link,
      linkLabel: this.slice?.primary?.linkLabel,
      image: this.slice?.primary?.image,
      linkedCompanyLogo: this.slice?.primary?.linkedCompanyLogo,
      subPosts: this.slice?.items,
      animation: this.slice?.primary?.animation,
    }
  },

  mounted() {
  },

  methods: {
    sendPressCenterPostEvent() {
      pressCenterClickEvent.send()
      pressCenterClickPixelEvent.send()
    },

    shortDescription(text) {
      const limit = 150
      return textEllipsis(text, { limit })
    },

    shortDescriptionForSubPosts(text) {
      const limit = 90
      return textEllipsis(text, { limit })
    },

    formattedDate(date) {
      return formatDate(date)
    },

  },

}

</script>

<style lang="scss" scoped>
.press-center-section {
  background-color: $bgcolor--white-primary;
}

//primary section styles
.ps-main-card {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  width: 100%;

  &__footer {
    margin-top: auto;
    display: flex;
    justify-content: space-between;
    align-items: baseline;
  }

  &__link {
    color: $text-color--black;
  }

  &-section {
    width: 50%;
    max-height: 285px;
    @media screen and (max-width: 1200px) {
      max-height: 330px;
    }
    @media screen and (min-width: 540px) and (max-width: 767px) {
      max-height: 285px;
    }
    @media screen and (max-width: 767px) {
      width: 100%;
      max-width: unset;
    }

    &--left {
      background-color: $bgcolor--white;
      box-sizing: border-box;
      padding: 40px 60px 35px;
      display: flex;
      flex-direction: column;
      max-height: 285px;
      @media screen and (min-width: 768px) and (max-width: 1200px) {
        padding: 20px 30px;
      }
      @media screen and (max-width: 767px) {
        padding: 18px;
        order: 2;
        height: 300px;
      }
    }

    &--right {
      box-sizing: border-box;
      overflow: hidden;
      position: relative;
      @media screen and (max-width: 767px) {
        order: 1;
        max-height: 200px;
      }
    }
  }

  &__title {
    @include font('Poppins', 26px, 700);
    line-height: 32px;
    color: $text-color--black;
    margin-bottom: 15px;
    @media screen and (max-width: 991px) {
      font-size: 18px;
      line-height: 24px;
    }
    @media screen and (max-width: 767px) {
      font-size: 22px;
      line-height: 30px;
    }
  }

  &__description {
    @include font('Poppins', 14px, 400);
    color: $text-color--grey-pale;
    line-height: 20px;
    letter-spacing: -0.4px;
    margin-bottom: 10px;
    @media screen and (max-width: 991px) {
      font-size: 12px;
      line-height: 18px;
    }
    @media screen and (max-width: 767px) {
      font-size: 14px;
      line-height: 23px;
    }
  }

  &__date {
    @include font('Poppins', 12px, 400);
    color: $text-color--grey;
  }

  &__link-label {
    @include font('Poppins', 16px, 600);

    &:hover {
      color: $text-color--red;
    }
  }

  &__image {
    width: 100%;
    height: 100%;
    object-fit: cover;
  }

  &__linked-logo {
    position: absolute;
    right: 30px;
    bottom: 25px;
    height: 30px;
    width: auto;
    @media screen and (max-width: 1200px) {
      height: 20px;
      right: 10px;
      bottom: 10px;
    }
    @media screen and (max-width: 767px) {
      height: 25px;
    }
  }
}

//primary section styles end
// -------------------------
//repeatable section styles

.ps-sub-cards {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  width: 100%;
  margin-top: 20px;
}

.ps-sub-card {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  box-sizing: border-box;
  width: 100%;
  height: 100%;

  &-wrapper {
    width: 50%;
    box-sizing: border-box;
    margin-bottom: 24px;
    @media screen and (min-width: 992px) {
      &:nth-child(odd) {
        padding-right: 24px;
      }
    }
    @media screen and (max-width: 992px) {
      width: 100%;
    }
  }

  &__footer {
    margin-top: auto;
    display: flex;
    justify-content: space-between;
    align-items: baseline;
  }

  &__link {
    color: $text-color--black;
  }

  &-section {
    width: 50%;
    max-height: 285px;
    @media screen and (max-width: 767px) {
      width: 100%;
      max-width: unset;
    }

    &--left {
      box-sizing: border-box;
      overflow: hidden;
      position: relative;
    }

    &--right {
      background-color: $bgcolor--white;
      box-sizing: border-box;
      padding: 18px;
      display: flex;
      flex-direction: column;
      @media screen and (max-width: 767px) {
        height: 300px;
        padding: 15px 14px 14px 14px;
      }
    }
  }

  &__image {
    width: 100%;
    height: 100%;
    object-fit: cover;
  }

  &__linked-logo {
    position: absolute;
    right: 20px;
    bottom: 20px;
    height: 20px;
    width: auto;
    @media screen and (max-width: 1200px) {
      height: 20px;
      right: 10px;
      bottom: 10px;
    }
    @media screen and (max-width: 767px) {
      height: 25px;
    }
  }

  &__title {
    @include font('Poppins', 18px, 700);
    line-height: 24px;
    color: $text-color--black;
    margin-bottom: 10px;
    @media screen and (max-width: 991px) {
      font-size: 12px;
      line-height: 18px;
    }
    @media screen and (max-width: 767px) {
      @include font('Poppins', 18px, 600);
      font-size: 23px;
      line-height: 30px;
    }
  }

  &__description {
    @include font('Poppins', 12px, 400);
    color: $text-color--grey-pale;
    line-height: 18px;
    margin-bottom: 18px;
    @media screen and (max-width: 991px) {
      margin-bottom: 20px;
    }
    @media screen and (max-width: 767px) {
      font-size: 14px;
      line-height: 23px;
    }
  }

  &__date {
    @include font('Poppins', 12px, 400);
    color: $text-color--grey;
  }

  &__link-label {
    @include font('Poppins', 16px, 600);

    &:hover {
      color: $text-color--red;
    }
  }
}

</style>
