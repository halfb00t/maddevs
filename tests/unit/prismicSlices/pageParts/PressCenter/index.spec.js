import { fireEvent, render, screen } from '@testing-library/vue'
import lazyLoad from 'nuxt-lazy-load'
import { createLocalVue, shallowMount } from '@vue/test-utils'
import PressCenter from '@/prismicSlices/pageParts/PressCenter'
import { pressCenterClickEvent } from '@/analytics/events'

const localVue = createLocalVue()
localVue.directive('lazy-load', lazyLoad)
jest.mock('nuxt-lazy-load/lib/module')
const props = {
  slice: {
    primary: {
      title: 'test',
      description: 'test desc',
      date: new Date().toISOString(),
      link: {
        url: 'https://maddevs.io',
      },
      linkLabel: 'Test',
      image: {
        url: 'img.jpg',
        alt: 'Test',
      },
      linkedCompanyLogo: {
        url: 'img-logo.jpg',
        alt: 'Test logo',
      },
      animation: 'fade-up',
    },
    items: [
      {
        image: {
          url: 'img.jpg',
          alt: 'Test alt',
        },
        title: 'Test Test',
        description: 'Test description',
        link: {
          url: 'https://maddevs.co',
        },
        date: new Date().toISOString(),
        linkLabel: 'Test label',
        linkedCompanyLogo: {
          url: 'img-logo.jpg',
          alt: 'Test',
        },
      },
    ],
  },
}
jest.mock('~/helpers/generatorUid')
const pressCenterClick = jest.spyOn(pressCenterClickEvent, 'send')
  .mockImplementation(() => {
  })
describe('PressCenter component', () => {
  it('should correct render component', async () => {
    const { container } = render(PressCenter, {
      propsData: props,
      localVue,
    })
    const link = screen.getByTestId('press-center-link')
    await fireEvent.click(link)
    expect(pressCenterClick).toHaveBeenCalledTimes(1)
    expect(container).toMatchSnapshot()
  })
  it('should return empty object from slice props', () => {
    const wrapper = shallowMount(PressCenter, {
      propsData: props,
      localVue,
    })
    expect(wrapper.vm.$options.props.slice.default.call()).toEqual({})
  })
})
