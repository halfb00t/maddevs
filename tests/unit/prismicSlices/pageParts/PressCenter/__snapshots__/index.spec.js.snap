// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PressCenter component should correct render component 1`] = `
<div>
  <section
    class="section press-center-section"
  >
    <div
      class="container"
    >
      <div
        class="press-center"
        data-aos="fade-up"
      >
        <div
          class="press-center-main ps-main-card"
        >
          <div
            class="ps-main-card-section ps-main-card-section--left"
          >
            <h2
              class="ps-main-card__title"
            >
              
            test
          
            </h2>
             
            <div
              class="ps-main-card__description"
            >
              
            test desc
          
            </div>
             
            <div
              class="ps-main-card__footer"
            >
              <span
                class="ps-main-card__date"
              >
                Feb 03, 2020
              </span>
               
              <a
                class="ps-main-card__link"
                data-testid="press-center-link"
                href="https://maddevs.io"
                target="_blank"
              >
                <span
                  class="ps-main-card__link-label"
                >
                  
                Test
              
                </span>
              </a>
            </div>
          </div>
           
          <div
            class="ps-main-card-section ps-main-card-section--right"
          >
            <img
              alt="Test"
              class="ps-main-card__image"
              data-src="img.jpg"
            />
             
            <img
              alt="Test logo"
              class="ps-main-card__linked-logo"
              data-src="img-logo.jpg"
            />
          </div>
        </div>
         
        <div
          class="press-center-sub ps-sub-cards"
        >
          <div
            class="press-center-sub-wrapper ps-sub-card-wrapper"
          >
            <div
              class="press-center-sub ps-sub-card"
            >
              <div
                class="ps-sub-card-section ps-sub-card-section--left"
              >
                <img
                  alt="Test alt"
                  class="ps-sub-card__image"
                  data-src="img.jpg"
                />
                 
                <img
                  alt="Test"
                  class="ps-sub-card__linked-logo"
                  data-src="img-logo.jpg"
                />
              </div>
               
              <div
                class="ps-sub-card-section ps-sub-card-section--right"
              >
                <h2
                  class="ps-sub-card__title"
                >
                  
                Test Test
              
                </h2>
                 
                <div
                  class="ps-sub-card__description"
                >
                  
                Test description
              
                </div>
                 
                <div
                  class="ps-sub-card__footer"
                >
                  <span
                    class="ps-sub-card__date"
                  >
                    Feb 03, 2020
                  </span>
                   
                  <a
                    class="ps-sub-card__link"
                    href="https://maddevs.co"
                    target="_blank"
                  >
                    <span
                      class="ps-sub-card__link-label"
                    >
                      
                    Test label
                  
                    </span>
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;
